data_steward/cdr_cleaner/cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,250 @@
+        for _, table_info in table_info_df.iterrows():
+            cols = table_info['columns']
+            concept_fields = [dict(name=col, alias=col.replace('_id', '')) for col in cols if is_concept_id_col(col)]
+            if concept_fields:
+                # Otherwise there are no rows to remove
+                sub_query = ROWS_TO_SUPPRESS_QUERY.render(
+                    sandbox_table_id=concept_lookup_table,
+                    dataset_id=dataset_id,
+                    table=table_info['table_name'],
+                    ext_table=table_info['ext_table_name'],
+                    concept_fields=concept_fields)
+                sub_queries.append(sub_query)
+        query = "\nUNION ALL\n".join(sub_queries)
+        dataset_queries.append(query)
+    return dataset_queries
+
+
+def get_rows_to_suppress_df(project_id, target_dataset_ids, concept_lookup_table):
+    """
+    Retrieve key info on rows to suppress from target datasets
+
+    :param project_id: identifies the project containing the target datasets
+    :param target_dataset_ids: list of dataset_ids of datasets to suppress
+    :param concept_lookup_table: identifies the lookup table with concepts to suppress
+    :return: dataframe with dataset_id, disallowed_concept_id, table, n_person_id, n_row_id, n_source, ppi_pm
+    """
+    dataset_queries = get_rows_to_suppress_queries(project_id, target_dataset_ids, concept_lookup_table)
+    return pd.concat(client.query(dataset_query).to_dataframe() for dataset_query in dataset_queries)
+
+
+def ppi_pm(g):
+    """
+    True if any rows in the group have source PPI/PM
+    """
+    return g['source'].isin(['PPI/PM']).any()
+
+
+def get_suppress_summary_df(rows_to_suppress_df):
+    """
+    Retrieve summary info (counts) on rows to suppress from target datasets
+
+    :param rows_to_suppress_df: info on rows to suppress from target datasets
+    :return: dataframe with dataset_id, table, disallowed_concept_id, person_id, row_id, source
+    """
+    summary_cols = ['dataset_id', 'table', 'disallowed_concept_id', 'person_id', 'row_id', 'source']
+    group_by_concept = rows_to_suppress_df[summary_cols].groupby(['dataset_id', 'table', 'disallowed_concept_id'])
+    summary_df = group_by_concept.nunique().drop(columns=['dataset_id', 'table', 'disallowed_concept_id'])
+    summary_df = summary_df.add_prefix('n_')
+    summary_df['ppi_pm'] = group_by_concept.apply(ppi_pm)
+    return summary_df.sort_values(by=['dataset_id', 'table', 'n_person_id'], ascending=False)
+
+
+def main(project_id, sandbox_dataset_id, concept_lookup_table, target_dataset_ids):
+    rows_dest_table = f'{sandbox_dataset_id}.{ROWS_RESOURCE_NAME}'
+    suppress_rows_df = get_rows_to_suppress_df(project_id, target_dataset_ids, concept_lookup_table)
+    suppress_rows_df.to_csv(f'{ROWS_RESOURCE_NAME}.csv')
+    suppress_rows_df.to_gbq(destination_table=rows_dest_table, if_exists='replace')
+    suppress_summary_df = get_rows_to_suppress_df(SUPPRESS_ROWS_DF)
```

Comment:
should this be `get_suppress_summary_df` ??

Reply:
Yes, thanks, this has been fixed